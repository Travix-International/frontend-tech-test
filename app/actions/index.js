import {FILTERS, TODO_ACTIONS, FETCH } from './types';

let nextTodoId = 0;
export const addNewTodo = text => {
    return {
        type: TODO_ACTIONS.ADD_TODO,
        id: nextTodoId++,
        text
    }
};

export const setVisibilityFilter = filter => {
    return {
        type: FILTERS.FILTER_BY,
        filter
    }
};

export const toggleTodo = id => {
    return {
        type: FILTERS.TOGGLE_TODO,
        id
    }
};


function requestTasks(tasks) {
    return {
        type: FETCH.GET_TASKS,
        tasks
    }
}