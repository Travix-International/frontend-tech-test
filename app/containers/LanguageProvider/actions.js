import { actionTypes as at } from './constants';

export const changeLocale = languageLocale => ({
  type: at.CHANGE_LOCALE,
  locale: languageLocale
});
