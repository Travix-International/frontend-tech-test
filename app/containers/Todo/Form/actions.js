import { actionTypes as at } from './constants';

export const change = ({ value, name }) => ({
  type: at.CHANGE,
  payload: { value, name }
});

export const save = () => ({
  type: at.SAVE
});

export const saveSuccess = ({ data }) => ({
  type: at.SAVE_SUCCESS,
  payload: { data }
});

export const saveError = () => ({
  type: at.SAVE_ERROR
});
