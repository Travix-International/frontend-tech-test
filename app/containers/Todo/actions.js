import { actionTypes as at } from './constants';

export const fetch = () => ({
  type: at.FETCH
});

export const fetchSuccess = ({ data }) => ({
  type: at.FETCH_SUCCESS,
  payload: { data }
});

export const fetchError = ({ error }) => ({
  type: at.FETCH_ERROR,
  payload: { error }
});

export const update = ({ id, description, title, index }) => ({
  type: at.UPDATE,
  payload: {
    id,
    description,
    title,
    index
  }
});

export const updateSuccess = ({ data, index }) => ({
  type: at.UPDATE_SUCCESS,
  payload: { data, index }
});

export const updateError = ({ id, index }) => ({
  type: at.UPDATE_ERROR,
  payload: { id, index }
});

export const remove = ({ id, index }) => ({
  type: at.REMOVE,
  payload: { id, index }
});

export const removeSuccess = ({ data, index }) => ({
  type: at.REMOVE_SUCCESS,
  payload: { data, index }
});

export const removeError = ({ id, index }) => ({
  type: at.REMOVE_ERROR,
  payload: { id, index }
});
