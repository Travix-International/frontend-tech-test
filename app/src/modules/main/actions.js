import {
  TASK_REQUEST
} from './../../constants';

export const requestTask = () => ({
  type: TASK_REQUEST
});

export default requestTask;