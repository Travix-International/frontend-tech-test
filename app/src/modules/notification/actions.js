import {
  NOTIFICATION
 } from './../../constants';

export const notificationAction = (show, success, message) => ({
  type: NOTIFICATION,
  show,
  success,
  message
});

export default notificationAction;