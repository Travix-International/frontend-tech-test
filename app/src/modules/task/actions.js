import 'core-js/fn/promise';
import 'whatwg-fetch';
import {
  TASK_LIST,
  TASK_DELETE
} from './../../constants';
import { notificationAction } from './../notification/actions';
import { requestTask } from './../main/actions';
import { config } from './../../config';

export const taskListAction = tasks => ({
  type: TASK_LIST,
  tasks,
});

export const taskDeleteAction = id => ({
  type: TASK_DELETE,
  id
});

export const taskList = () => (dispatch) => {
  dispatch(requestTask());
  const options = {
    method: 'GET'
  };
  const url = `${config.api}task`;

  fetch(url, options)
    .then(res => res.json())
    .then((json) => {
      dispatch(taskListAction(json));
    }).catch(() => {
      dispatch(taskListAction(null));
    });
};

export const taskDelete = id => (dispatch) => {
  dispatch(requestTask());

  const url = `${config.api}task/${id}`;
  const options = {
    method: 'DELETE'
  };

  fetch(url, options)
    .then(res => res.json())
    .then((json) => {
      const success = (json.ok > 0);
      const message = (json.ok > 0 ? 'Task removed!' : 'Occured a problem, try again!');
      dispatch(notificationAction(true, success, message));
      dispatch(taskDeleteAction(id));
      setTimeout(() => {
        dispatch(notificationAction(false, false, null));
      }, 5000);
    });
};
