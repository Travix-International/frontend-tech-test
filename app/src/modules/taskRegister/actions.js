import 'core-js/fn/promise';
import 'whatwg-fetch';
import {
  TASK_SAVE
} from './../../constants';
import { notificationAction } from './../notification/actions';
import { requestTask } from './../main/actions';
import { config } from './../../config';
import { history } from './../../store';

const taskSaveAction = (_id, title, description, date, completed) => ({
  type: TASK_SAVE,
  _id,
  title,
  description,
  date,
  completed
});

export const taskSave = (id, title, description, date, completed) => (dispatch) => {
  dispatch(requestTask());
  const body = {
    title,
    description,
    date,
    completed
  };
  const options = {
    method: (id && id !== '0' ? 'PUT' : 'POST'),
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body)
  };
  const url = (id && id !== '0' ? `${config.api}task/${id}` : `${config.api}task`);

  fetch(url, options)
    .then(res => res.json())
    .then((json) => {
      const success = (json._id != null);
      const message = (json._id != null ? 'Task saved!' : 'Occurred a problem, try again!');
      dispatch(notificationAction(true, success, message));
      dispatch(taskSaveAction(json._id, title, description, date, completed));
      history.push('/');
      setTimeout(() => {
        dispatch(notificationAction(false, false, null));
      }, 5000);
    });
};

export default taskSave;
