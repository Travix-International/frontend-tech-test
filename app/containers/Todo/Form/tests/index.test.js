import React from 'react';
import { create } from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import { TodoForm, mapDispatchToProps } from '../index';
import { change, save } from '../actions';

const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <IntlProvider {...props}>
      <TodoForm {...finalProps} />
    </IntlProvider>
  );
};

describe('<TodoForm />', () => {
  it('should render component', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChange', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChange).toBeDefined();
      });

      it('should dispatch change when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const value = 'value';
        const name = 'name';

        result.onChange(value, name);

        expect(dispatch).toHaveBeenCalledWith(change({ value, name }));
      });
    });

    describe('onSubmit', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        expect(result.onSubmit).toBeDefined();
      });

      it('should dispatch save when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        result.onSubmit();

        expect(dispatch).toHaveBeenCalledWith(save());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };

        result.onSubmit(evt);

        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
