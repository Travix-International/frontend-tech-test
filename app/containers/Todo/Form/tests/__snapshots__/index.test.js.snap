// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> should render component 1`] = `
<form
  className="Form__Form-PASBS rsOuw"
  onSubmit={undefined}
>
  <label
    htmlFor="title"
  >
    <span
      className="Wrapper-dyuznp ejfftc"
    >
      <input
        className="StyledInput__Input-eYzgai fmVHMz"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value={undefined}
      />
      <label
        className="Label-ghyrbC bcBwfc"
      >
        <label
          className="LabelContent-iptDuP ccpyDu"
        >
          Title
        </label>
      </label>
    </span>
  </label>
  <label
    htmlFor="description"
  >
    <span
      className="Wrapper-dyuznp ejfftc"
    >
      <input
        className="StyledInput__Input-eYzgai fmVHMz"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value={undefined}
      />
      <label
        className="Label-ghyrbC bcBwfc"
      >
        <label
          className="LabelContent-iptDuP ccpyDu"
        >
          Description
        </label>
      </label>
    </span>
  </label>
  <div
    className="SubmitButton__SubmitButton-iZeMNu fhFgOJ Wrapper__Wrapper-dXHlXX ggGEtw"
  >
    <button
      className="StyledButton__StyledButton-iQSVZX hrAcih"
      disabled={true}
      onClick={undefined}
      type="submit"
    >
      <span>
        Save
      </span>
    </button>
  </div>
</form>
`;
